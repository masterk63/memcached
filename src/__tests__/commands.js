const { runCommand } = require('../commands');
const Socket = require('../models/Socket');
const memcached = require('../memcached');
const cache = memcached.getMemcachedInstance();

jest.mock('../models/Socket');

const parseCommand = command => command.split(' ');

beforeEach(() => {
  Socket.mockClear();
});

describe('Store Commands', () => {
  test('Set command bad chunk', () => {
    const socket = new Socket();
    const command = 'set foo 3 3 3';
    const value = 'done';
    runCommand(socket, parseCommand(command), value);
    expect(socket.badDataChunk).toHaveBeenCalledTimes(1);
    expect(socket.storedMessage).not.toHaveBeenCalled();
    expect(memcached.isKeyStored('foo')).toBe(false);
  });
  test('Set command ran successfully', () => {
    const socket = new Socket();
    const command = 'set foo 3 3 3';
    const value = 'say';
    runCommand(socket, parseCommand(command), value);
    expect(socket.storedMessage).toHaveBeenCalledTimes(1);
    expect(socket.badDataChunk).not.toHaveBeenCalled();
    expect(JSON.stringify(cache)).toMatchSnapshot();  
    memcached.deleteKeyCache('foo');  
  });
  test('Set command with exptime -1 (auto delete)', () => {
    const socket = new Socket();
    const command = 'set foo 3 -1 3';
    const value = 'say';
    runCommand(socket, parseCommand(command), value);
    expect(socket.storedMessage).toHaveBeenCalledTimes(1);
    expect(socket.badDataChunk).not.toHaveBeenCalled();
    expect(memcached.isKeyStored('foo')).toBe(false);
  });
});
