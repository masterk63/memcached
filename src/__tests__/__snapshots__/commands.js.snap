// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store Commands Set command ran successfully 1`] = `"{\\"foo\\":{\\"key\\":\\"foo\\",\\"value\\":\\"say\\",\\"flags\\":3,\\"cas\\":1,\\"fetchLog\\":[],\\"updateLog\\":[{\\"time\\":1580414508505}]}}"`;
